### ISSUE TITLE
Good certificates/private key will be declared as invalid

### ISSUE THREAD

**Issue by ph** [logstash-plugins/logstash-input-beats#123]
The integration test should have pick this up, I believe the latest changes from #119 broke the plugin. There seems to be a problem with the handling of private key and the converter, it is used to convert from PKC7 to PKCS8.

```
Aug 25, 2016 9:40:00 AM io.netty.channel.ChannelInitializer exceptionCaught
WARNING: Failed to initialize a channel. Closing: [id: 0x3759a880, L:/127.0.0.1:5044 - R:/127.0.0.1:63831]
java.lang.IllegalArgumentException: Input stream not contain valid certificates.
	at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:272)
	at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:99)
	at org.logstash.netty.SslSimpleBuilder.build(SslSimpleBuilder.java:107)
	at org.logstash.beats.Server$BeatsInitializer.initChannel(Server.java:132)
	at org.logstash.beats.Server$BeatsInitializer.initChannel(Server.java:99)
	at io.netty.channel.ChannelInitializer.channelRegistered(ChannelInitializer.java:68)

**Comment by ph** [comment]
This error was detected when running the `Runner` class inside the java project, 2.4 doesn't expose that much information.

**Comment by tsg** [comment]
Still failing for me with plugin version 3.1.1, not sure if I'm doing anything else wrong. I don't get any errors in the Logstash logs, but I uploaded the full logs [here](https://gist.github.com/tsg/5e8d1e9db0fe84bd393d165457f0d885) in case I'm missing something.

**Comment by ph** [comment]
Fixed in 3.1.2. Make sure the private keys are in PKCS8.
